mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergInputFormat.java

```diff
@@ -113,17 +93,23 @@ public RecordReader<Void, Container<Record>> getRecordReader(
 
     if (HiveConf.getBoolVar(job, HiveConf.ConfVars.HIVE_VECTORIZATION_ENABLED)
         && Utilities.getVectorizedRowBatchCtx(job) != null) {
-      Preconditions.checkArgument(
-          HiveVersion.min(HiveVersion.HIVE_3), "Vectorization only supported for Hive 3+");
 
       job.setEnum(InputFormatConfig.IN_MEMORY_DATA_MODEL, InputFormatConfig.InMemoryDataModel.HIVE);
       job.setBoolean(InputFormatConfig.SKIP_RESIDUAL_FILTERING, true);
 
       IcebergSplit icebergSplit = ((IcebergSplitContainer) split).icebergSplit();
+      DynConstructors.Ctor<AbstractMapredIcebergRecordReader> ctor =
```

Comment:
Probably it would be better to still keep a constant for this instead of recreating it every time